src/app/src/main/java/com/cmput301f21t09/budgetprojectname/HabitEventController.java

```diff
@@ -0,0 +1,221 @@
+        void onCallback(String habitEventID);
+    }
+
+    public interface HabitEventListCallback {
+        void onCallback(ArrayList<HabitEventModel> habitEventList);
+    }
+
+
+    /**
+     * Creates habitEvent document in the habit_events collection
+     *
+     * @param habitEvent habitEvent to be added
+     */
+    public void createHabitEvent(HabitEventModel habitEvent, HabitEventIDCallback idCallback) {
+        // db = FirebaseFirestore.getInstance();
+        System.out.println("store new habit");
+        dbStore.collection("habit_events")
+                .add(habitEvent)
+                .addOnSuccessListener(new OnSuccessListener<DocumentReference>() {
+                    @Override
+                    public void onSuccess(DocumentReference documentReference) {
+                        String docID = documentReference.getId();
+                        Log.d(TAG, "DocumentSnapshot added with ID: " + docID);
+                        System.out.println(docID);
+                        idCallback.onCallback(docID);
+                    }
+                })
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        Log.w(TAG, "Error adding document", e);
+                    }
+                });
+    }
+
+    /**
+     * Updates habitEvent document in the habit_events collection
+     *
+     * @param habitEventID       ID of habitEvent to be updated
+     * @param modifiedHabitEvent habitEvent to be updated
+     */
+    public void updateHabitEvent(String habitEventID, HabitEventModel modifiedHabitEvent) {
+        DocumentReference habitEventRef = dbStore.collection("habit_events")
+                .document(habitEventID);
+        habitEventRef.update("comment", modifiedHabitEvent.getComment(),
+                "location", modifiedHabitEvent.getLocation(),
+                "image", modifiedHabitEvent.getImage())
+                .addOnSuccessListener(new OnSuccessListener<Void>() {
+                    @Override
+                    public void onSuccess(Void aVoid) {
+                        Log.d(TAG, "DocumentSnapshot successfully updated!");
+                    }
+                })
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        Log.w(TAG, "Error updating document", e);
+                    }
+                });
+    }
+    // updateLoc()
+    /**
+     * new HEModel(oldHE stuff, new loc)
+     * cachedModel = heModel
+     * notifyListeners()
+     * commit()
+     */
+
+    /**
+     * Read habitEvent document in the habit_events collection
+     *
+     * @param habitID habitID is used to query all the corresponding habit events
+     */
+    public void readHabitEvent(String habitID, HabitEventListCallback hbEvtLstCallback) {
```

Comment:
Ah sorry about this but because we changed from final fields to modifieable fields, I ended up just having one constructor. Could you reduce this method to just the one constructor which may have some null fields?

Reply:
I fixed it in the latest commit!

Reply:
what i meant was i only have 1 habiteventmodel constructor now and i think your code is using different ones depending on the provided params

Reply:
Yep! Here you go: https://github.com/CMPUT301F21T09/BudgetProjectName/pull/122/commits/1b94d74c08a91f059401342d86718354e39b23ec